File: Python/1880.py
input()
first = list(map(int, input().split()))
input()
second = list(map(int, input().split()))
input()
third = list(map(int, input().split()))

count = 0
fi, si, ti = 0, 0, 0
while fi < len(first) and si < len(second) and ti < len(third):
    a = first[fi]
    b = second[si]
    c = third[ti]
    if a == b and b == c:
        count += 1
        fi += 1
        si += 1
        ti += 1
    elif a <= b and a <= c:
        fi += 1
    elif b <= a and b <= c:
        si += 1
    else:
        ti += 1
print(count)

File: C++/1880.cpp
#include <algorithm>
#include <cstddef>
#include <iostream>
#include <iterator>
#include <vector>

int main() {
  std::size_t size;
  std::vector<int> first;
  std::vector<int> second;
  std::vector<int> third;

  std::cin >> size;
  first.reserve(size);
  std::copy_n(std::istream_iterator<int>(std::cin), size,
              std::back_inserter(first));
  std::cin >> size;
  second.reserve(size);
  std::copy_n(std::istream_iterator<int>(std::cin), size,
              std::back_inserter(second));
  std::cin >> size;
  third.reserve(size);
  std::copy_n(std::istream_iterator<int>(std::cin), size,
              std::back_inserter(third));

  int count = 0;
  auto fb = first.begin();
  auto fe = first.end();
  auto sb = second.begin();
  auto se = second.end();
  auto tb = third.begin();
  auto te = third.end();
  while (fb != fe && sb != se && tb != te) {
    auto a = *fb;
    auto b = *sb;
    auto c = *tb;
    if (a == b && b == c) {
      ++count;
      ++fb;
      ++sb;
      ++tb;
    } else if (a <= b && a <= c) {
      ++fb;
    } else if (b <= a && b <= c) {
      ++sb;
    } else {
      ++tb;
    }
  }
  std::cout << count << '\n';
  return 0;
}
